src/components/ClickableStyle/ClickableStyle.tsx

```diff
@@ -0,0 +1,134 @@
+  /**
+   * On click handler for component
+   */
+  onClick?: MouseEventHandler;
+  /**
+   * Available size variations for the clickable
+   */
+  size?: 'sm' | 'md' | 'lg';
+  /**
+   * The visible clickable children
+   */
+  children: string | ReactNode;
+  /**
+   * Determines type of clickable
+   * - **button** The clickable is a clickable button.
+   * - **submit** The clickable is a submit clickable (submits form data).
+   * - **reset** The clickable is a reset clickable (resets the form-data to its initial values)
+   */
+  type?: 'button' | 'reset' | 'submit';
+  /**
+   * Available _stylistic_ variations available for the ClickableStyle component
+   */
+  variant?:
+    | 'primary'
+    | 'secondary'
+    | 'tertiary'
+    | 'icon'
+    | 'link'
+    | 'destructive';
+} & React.ComponentProps<IComponent>;
+
+/**
+ * A helper component that contains all the styling for buttons and links.
+ *
+ * If you're styling a `<button>` or `<a>` element, you can use the `Button`
+ * and `Link` components (respectively). `ClickableStyle` should only be used
+ * directly when styling other elements or components (e.g. `Link` from `react-router`)
+ * to look like a button or link.
+ *
+ * See the `Button` and `Link` stories for usage examples.
+ */
+export const ClickableStyle = React.forwardRef(
+  <IComponent extends React.ElementType>(
+    {
+      as: Component,
+      className,
+      disabled,
+      forwardRef,
+      fullWidth,
+      loading,
+      size = 'lg',
+      children,
+      variant = 'primary',
+      ...other
+    }: ClickableStyleProps<IComponent>,
+    ref: React.ForwardedRef<HTMLElement>,
+  ) => {
+    const componentClassName = clsx(
+      // Base styles
+      styles['clickable-style'],
+      className,
+      // Sizes
+      variant !== 'link' && [
+        size === 'sm' && styles['clickable-style--sm'],
+        size === 'md' && styles['clickable-style--md'],
+        size === 'lg' && styles['clickable-style--lg'],
+      ],
+      // Variants
+      variant === 'primary' && styles['clickable-style--primary'],
+      variant === 'secondary' && styles['clickable-style--secondary'],
+      variant === 'tertiary' && styles['clickable-style--tertiary'],
+      variant === 'icon' && styles['clickable-style--icon'],
+      variant === 'link' && styles['clickable-style--link'],
+      variant === 'destructive' && styles['clickable-style--destructive'],
+      // Other options
+      fullWidth && styles['clickable-style--full-width'],
+      loading && styles['eds-is-loading'],
+    );
+
+    return (
+      <Component
+        className={componentClassName}
+        disabled={disabled}
+        ref={forwardRef || ref}
+        tabIndex={disabled ? -1 : undefined}
+        {...other}
+      >
+        {loading && (
+          <Icon
+            className={styles['clickable__icon-loading']}
+            name="spinner"
+            purpose="informative"
+            title="loading"
+            size="1.5em"
```

Comment:
This icon is really big compared to our other ones because it fills the whole bounding box. When we clean up the whitespace on the other icons, we'll probably want them all to be 1.5em instead of 2.